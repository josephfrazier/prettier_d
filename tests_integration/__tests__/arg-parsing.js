"use strict";

const runPrettier = require("../runPrettier");

describe("boolean flags do not swallow the next argument", () => {
  runPrettier("cli/arg-parsing", [
    "--end-of-line",
    "lf",
    "--single-quote",
    "file.js",
  ]).test({
    status: 0,
  });
});

describe("negated options work", () => {
  runPrettier("cli/arg-parsing", [
    "--end-of-line",
    "lf",
    "--no-semi",
    "file.js",
  ]).test({
    status: 0,
  });
});

describe("unknown options are warned", () => {
  runPrettier("cli/arg-parsing", [
    "--end-of-line",
    "lf",
    "file.js",
    "--unknown",
  ]).test({
    status: 0,
  });
});

describe("unknown negated options are warned", () => {
  runPrettier("cli/arg-parsing", [
    "--end-of-line",
    "lf",
    "file.js",
    "--no-unknown",
  ]).test({
    status: 0,
  });
});

describe("unknown options not suggestion `_`", () => {
  runPrettier("cli/arg-parsing", ["file.js", "-a"]).test({
    status: 0,
    write: [],
  });
});

describe("allow overriding flags", () => {
  runPrettier(
    "cli/arg-parsing",
    ["--tab-width=1", "--tab-width=3", "--parser=babel"],
    { input: "function a() { b }" }
  ).test({
    stdout: "function a() {\n   b;\n}\n",
    status: 0,
  });
});
