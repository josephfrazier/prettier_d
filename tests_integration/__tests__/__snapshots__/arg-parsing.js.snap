// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow overriding flags (stderr) 1`] = `""`;

exports[`allow overriding flags (write) 1`] = `Array []`;

exports[`boolean flags do not swallow the next argument (stderr) 1`] = `""`;

exports[`boolean flags do not swallow the next argument (stdout) 1`] = `
"console.log('could be single quote and without semi');
"
`;

exports[`boolean flags do not swallow the next argument (write) 1`] = `Array []`;

exports[`negated options work (stderr) 1`] = `""`;

exports[`negated options work (stdout) 1`] = `
"console.log(\\"could be single quote and without semi\\")
"
`;

exports[`negated options work (write) 1`] = `Array []`;

exports[`unknown negated options are warned (stderr) 1`] = `
"[warn] Ignored unknown option --no-unknown.
"
`;

exports[`unknown negated options are warned (stdout) 1`] = `
"console.log(\\"could be single quote and without semi\\");
"
`;

exports[`unknown negated options are warned (write) 1`] = `Array []`;

exports[`unknown options are warned (stderr) 1`] = `
"[warn] Ignored unknown option --unknown.
"
`;

exports[`unknown options are warned (stdout) 1`] = `
"console.log(\\"could be single quote and without semi\\");
"
`;

exports[`unknown options are warned (write) 1`] = `Array []`;

exports[`unknown options not suggestion \`_\` (stderr) 1`] = `
"[warn] Ignored unknown option -a. Did you mean -c?
"
`;

exports[`unknown options not suggestion \`_\` (stdout) 1`] = `
"console.log(\\"could be single quote and without semi\\");
"
`;
